### PR TITLE
Replace argument operands with arg instructions.

### PR DESCRIPTION
**Pull request by ptersilie**
Requires: https://github.com/ykjit/ykllvm/pull/176

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 1]
```diff
@@ -425,11 +436,8 @@ impl<'a> TraceBuilder<'a> {
                 let load = jit_ir::LookupGlobalInst::new(self.handle_global(*gd_idx)?)?;
                 self.jit_mod.push_and_make_operand(load.into())
             }
-            aot_ir::Operand::Arg { arg_idx, .. } => {
-                // Lookup the JIT instruction that was passed into this function as an argument.
-                // Unwrap is safe since an `Arg` means we are currently inlining a function.
-                // FIXME: Is the above correct? What about args in the control point frame?
-                Ok(self.frames.last().unwrap().args[usize::from(*arg_idx)].clone())
+            aot_ir::Operand::Arg { .. } => {
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:439]: So we don't completely remove `Operand::Arg`?

Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:439]: Yes, I realised that too now. Working on it.

Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/trace_builder.rs]: Try [563e094](https://github.com/ykjit/yk/pull/1261/commits/563e0942d0190c57c947ceca6931e6dc7588a4a6)

**Comment by vext01** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.

**Comment by ptersilie** [comment]
This won't work without https://github.com/ykjit/ykllvm/pull/176

**Comment by ptersilie** [comment]
Force pushed an update to the submodule. This is good to go now.

**Pull request merged**